### PR TITLE
Redis Integration Tests

### LINKED ISSUE #179: Add tests for the Redis trigger

### PR DESCRIPTION
**Pull request by ecumene**
closes #179

### CODE REVIEW

**Comment by lann** [comment]
@fibonacci1729

File: crates/redis/src/tests.rs [hunk 1]
```diff
@@ -32,11 +43,8 @@ async fn test_pubsub() -> Result<()> {
     let trigger: RedisTrigger =
         build_trigger_from_app(app, None, FollowComponents::None, None).await?;
 
-    // TODO
-    // use redis::{FromRedisValue, Msg, Value};
-    // let val = FromRedisValue::from_redis_value(&Value::Data("hello".into()))?;
-    // let msg = Msg::from_value(&val).unwrap();
-    // trigger.handle(msg).await?;
+    let msg = create_trigger_event("messages", "hello");
+    let res = trigger.handle(msg).await?;
```
Comment (lann) [inline crates/redis/src/tests.rs:47]:
```suggestion
    let _res = trigger.handle(msg).await?;
```

**Review by lann — COMMENTED** [review]
One nit to make the linter happy, but otherwise this seems like a clear improvement. :+1:

File: crates/redis/src/tests.rs [hunk 2]
```diff
@@ -16,6 +17,16 @@ pub(crate) fn init() {
...
+        Value::Data("message".into()),
+        Value::Data(channel.into()),
+        Value::Data(payload.into()),
+    ]))
+    .unwrap()
+}
+
+#[ignore]
 #[tokio::test]
 #[allow(unused)]
```
Comment (lann) [inline crates/redis/src/tests.rs:31]:
```suggestion
```

**Review by fibonacci1729 — APPROVED** [review]
Thank you @ecumene! This is great! Once @lann's comments are addressed im ok with merging this!

**Comment by ecumene** [comment]
Sweet, but is it enough for testing the trigger (i.e closes #179)

**Comment by lann** [comment]
I think ideally we would also check that the message echoed to stdout matches the input, but that would require more structural changes to the way stdio is managed in the engine.

File: crates/redis/src/tests.rs (hunk 2 above)
Comment (ecumene) [reply crates/redis/src/tests.rs:31]: good catch

**Comment by ecumene** [comment]
Note: I added `closes #179`, if that's okay with you guys.

**Review by fibonacci1729 — APPROVED** [review]

**Comment by ecumene** [comment]
Added the changes for the `Makefile` to include integration tests. If it doesn't pass CI, that's why!

**Comment by fibonacci1729** [comment]
Once CI finishes, I'll merge! Thanks again!!

**Pull request merged**